test/model/test/common/wiredtiger_util.cpp

```diff
@@ -0,0 +1,262 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "wiredtiger.h"
+extern "C" {
+#include "test_util.h"
+}
+
+#include "model/test/wiredtiger_util.h"
+#include "model/util.h"
+
+/*
+ * wt_get --
+ *     Read from WiredTiger.
+ */
+model::data_value
+wt_get(
+  WT_SESSION *session, const char *uri, const model::data_value &key, model::timestamp_t timestamp)
+{
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    const char *value;
+    char cfg[64];
+
+    value = nullptr;
+
+    if (timestamp == 0)
+        testutil_check(session->begin_transaction(session, nullptr));
+    else {
+        testutil_snprintf(cfg, sizeof(cfg), "read_timestamp=%" PRIx64, timestamp);
+        testutil_check(session->begin_transaction(session, cfg));
+    }
+    testutil_check(session->open_cursor(session, uri, nullptr, nullptr, &cursor));
+
+    model::set_wt_cursor_key(cursor, key);
+    testutil_check_error_ok(ret = cursor->search(cursor), WT_NOTFOUND);
```

Comment:
These functions are (so far) used only to test the model's behavior in non-transactional cases, which is why `WT_ROLLBACK` would be an unexpected error. But let me still allow it here, just in case it comes up in the future.